index.ts

```diff
@@ -53,11 +53,11 @@ function isEventTarget(elements: EventTarget | Document | ArrayLike<Element> | s
 /**
  * Delegates event to a selector.
  */
-function delegate<TElement extends Element = Element, TEvent extends Event = Event>(
+function delegate<TElement extends Element = Element, TEventType extends EventType = EventType>(
```

Comment:
Wouldn't _adding_ `EventType` have the same effect?

```suggestion
function delegate<
	TElement extends Element = Element,
	TEvent extends Event = GlobalEventHandlersEventMap[TEventType],
	TEventType extends EventType = EventType
>(
```

I'm not sure if this is right, but I've had generic parameters depend on each other

Reply:
I think the only drawback that this would have is that the event name now **has to** be part of `GlobalEventHandlersEventMap`, technically making it a breaking change anyway, but likely less common

Reply:
this is already the case today actually, as the `eventType` argument is already typed as `EventType` which is an alias for `keyof GlobalEventHandlersEventMap`

Reply:
I don't understand. "This" what specifically? By "today" you mean in this PR or `master`?

Reply:
I meant in master (but this stays true in that PR). That's why the readme has this section: https://github.com/fregante/delegate-it#custom-event-types-in-typescript

Reply:
Gotcha